src/rules/ec2/EC2IMDSv2.ts

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { parse } from 'path';
+import { CfnResource, Stack } from 'aws-cdk-lib';
+import { CfnInstance, CfnLaunchTemplate } from 'aws-cdk-lib/aws-ec2';
+import { NagRuleCompliance, NagRules } from '../../nag-rules';
+
+// Check to see if the EC2 Istance is configured with a launch profile
+//Ec2imdsv2
+
+/**
+ * Determine if the EC2 Instance has a launch profile
+ * @param node the CfnResource to check
```

Comment:
This rule also needs to check EC2 AutoScaling groups. In AutoScaling groups there are 2 situations to check for.  

1. When the autoscaling group uses [an EC2 launch template](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-launchtemplate.html)
2.  When using an [AutoScaling launch configuration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-launchconfiguration-metadataoptions.html)

Let's get the base EC2 instance case working properly before we address this

Reply:
The EC2IMDSv2 rule now checks Autoscaling